### PR TITLE
Use interactive_embedded_interpreter to dynamicly loading various third-party libraries

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66512

TLDR, we are able to use the interactive_embedded_interpreter (basically just torch::deploy interpreter with an interactive shell) to dynamicly load various third party libraries. We use the popular libraries numpy, scipy, regex, pandas for illustration purpose.

A couple of changes need to be done for the interactive_embedded_interpreter:
1, we need link with :embedded_interpreter_all rather than :embedded_interpreter so we can enable DEEPBIND and use our custom loader
2, we provide a pylibRoot path to construct the InterpreterManager. The path will be added to the embedded interpreter's sys.path. Typically we can pass in the python library root path in a conda environment so torch::deploy interpreter can find all installed packages.
3, we allow interactive_embedded_interpreter execute a script to ease recording the exploration of various python libraries.

Differential Revision: [D31587278](https://our.internmc.facebook.com/intern/diff/D31587278/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31587278/)!

### CODE REVIEW

**Review by suo — APPROVED** [review]
nice! Some small nits but otherwise looks good

**Comment by shunting314** [comment]
Clang-tidy job fail on torch/csrc/deploy/interactive_embedded_interpreter.cpp , complaining <glog/logging.h> not found. I guess we need add interactive_embedded_interpreter.cpp to the OSS build to fix that. I have a separate PR for that.
